external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/OffsetRange.scala

```diff
@@ -90,21 +90,23 @@ final class OffsetRange private(
     val topic: String,
     val partition: Int,
     val fromOffset: Long,
-    val untilOffset: Long) extends Serializable {
+    val untilOffset: Long,
+    val recordNumber: Long) extends Serializable {
```

Comment:
Does mima actually complain about binary compatibility if you just make recordNumber count?  It's just an accessor either way...

If so, and you have to do this, I'd name this recordCount consistently throughout.  Number could refer to a lot of things that aren't counts.